src/lib/operators/join_hash.hpp

```diff
@@ -22,6 +22,12 @@ class JoinHash : public AbstractJoinOperator {
  public:
   static bool supports(const JoinConfiguration config);
 
+  // The jobs that perform the actual materialize, partition by redix, build and probe are executed directly if the
```

Comment:
I'd write:
```The jobs that perform the actual materialization, radix partitioning, building,  and probing are added to the scheduler in case the number of elements to process is above `JOB_SPAWN_THRESHOLD`. If not, the job is executed directly.
This threshold needs to be re-evaluated over time to find the value which gives the best performance.